consensus/src/quorum_store/proof_coordinator.rs

```diff
@@ -35,17 +35,25 @@ pub(crate) enum ProofCoordinatorCommand {
 struct IncrementalProofState {
     info: SignedDigestInfo,
     aggregated_signature: BTreeMap<PeerId, bls12381::Signature>,
+    aggregated_voting_power: u128,
+    completed: bool,
 }
 
 impl IncrementalProofState {
     fn new(info: SignedDigestInfo) -> Self {
         Self {
             info,
             aggregated_signature: BTreeMap::new(),
+            aggregated_voting_power: 0,
```

Comment:
I think it makes sense now to create an SignatureAggregator utility, maybe even put it somewhere with ValidatorVerifier.rs and would also allow testing properly. Can return keys, do the linear algorithm with aggregated voting power/completed / recheck and everything, and would be easy to unit test too.

And actually having such tests, even basic ones to start, would be great?